### DISCUSSION TITLE
Does TCA support only iOS 17?

### DISCUSSION THREAD

**Discussion by ArturRuZ** [pointfreeco/swift-composable-architecture#2826] [Q&A] [answered]
Hi! I'm using Xcode Version 15.2 (15C500b) and my test project (from the start guide) works only on iOS 17+. It doesn't work on iOS lower 17. Is it the bug? Unfortunately, I can use only simulators for testing. 
[TCADemo.zip](https://github.com/pointfreeco/swift-composable-architecture/files/14322900/TCADemo.zip)

**Comment by kamcma** [comment, marked as answer]
TCA 1.x supports iOS 13. However, whether you support only iOS 17 or need support for iOS 13-16 determines whether you can use Apple's vanilla Observation tools or must use TCA's Observation backport called Perception.

Would highly recommend you read [this doc article on the Observation backport](https://pointfreeco.github.io/swift-composable-architecture/1.8.0/documentation/composablearchitecture/observationbackport).

**Reply by ArturRuZ** [reply]
Thank you! Amazing

**Comment by mbrandonw** [comment]
Hi @ArturRuZ, when I download your project it compiles just fine in iOS 16. What problem are you having?

**Reply by ArturRuZ** [reply]
Actions don't work. If you press on buttons nothing will happen

**Reply by mbrandonw** [reply]
I see. That is because when deploying to iOS <17 you must wrap your views in `WithPerceptionTracking` as @kamcma mentioned above.
